build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/rest/compat/YamlRestCompatTestPlugin.java

```diff
@@ -198,6 +198,7 @@ public void apply(Project project) {
         // setup the test task
         Provider<RestIntegTestTask> yamlRestCompatTestTask = RestTestUtil.registerTestTask(project, yamlCompatTestSourceSet, testTaskName);
         project.getTasks().withType(RestIntegTestTask.class).named(testTaskName).configure(testTask -> {
+            testTask.getSystemProperties().put("tests.restCompat", true);
```

Comment:
This can be simplified to just `testTask.systemProperty("tests.restCompat", "true")`